Automattic/wp-calypso
# Stats: Using site type detection to streamline purchase page

Related to #80846

## Proposed Changes

* Incorporate newly added site detection flag to streamline the purchase flow for stats
* Add a feature flag

## Testing Instructions



* verify that the purchase page behaviour has not changed when no feature flag is applied
* verify that the purchase page behaviour has not changed for a non-commercial page (without and with `stats/type-detection` feature flag)
* currently it's hard to test the flow for commercial pages because the detection has not yet been finalised - verify that for commercial pages the flow doesn't change or you are redirected straight to the commercial plan purchase screen

## Pre-merge Checklist
